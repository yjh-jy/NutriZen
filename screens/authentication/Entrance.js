import { StyleSheet, Text, View, SafeAreaView, TouchableOpacity, Image } from 'react-native'
import {useCallback} from 'react'
import { useFonts } from 'expo-font';
import * as SplashScreen from 'expo-splash-screen';
import colors from '../../assets/colors/colors'

SplashScreen.preventAutoHideAsync();

export default Entrance = ({navigation}) => {
    const [fontsLoaded] = useFonts({
    "PixeloidSan": require("../../assets/fonts/PixeloidSans-mLxMm.ttf"),
    "PixeloidsSanBold": require("../../assets/fonts/PixeloidSansBold-PKnYd.ttf"),
    "MinimalPixel": require("../../assets/fonts/MinimalPixelFont.ttf")
    });

    const onLayoutRootView = useCallback(async () => {
    if (fontsLoaded) {
        await SplashScreen.hideAsync();
    }
    }, [fontsLoaded]);

    if (!fontsLoaded) {
    return null;
    }

  return (
    <View style={styles.container} onLayout={onLayoutRootView} >
        <SafeAreaView>
            
            <View style = {styles.itemsWrapper}>
                <Text style = {styles.title}>NutriZen</Text>
                <Image 
                style = {styles.image} 
                source= {require("../../assets/images/creature_loading_non_looping.gif")}
                >
                </Image>
                <TouchableOpacity onPress={() => navigation.navigate('Login')}>
                    <Text style = {styles.login}>Login</Text>
                </TouchableOpacity>

                <TouchableOpacity onPress={() => navigation.navigate('Registration')}>
                    <Text style = {styles.register}>Register</Text>
                </TouchableOpacity>
            </View>

            </SafeAreaView>
    </View>
  )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection:'column',
        backgroundColor: colors.backgroundColor,
    },
    itemsWrapper: {
        marginTop: '60%',
        alignItems: 'center',
        justifyContent: 'center'

    },
    title: {
        fontFamily: 'PixeloidsSanBold',
        fontSize: 48,
    },
    image:{
        width:140,
        height:61,

    },
    login: {
        fontFamily: 'PixeloidsSanBold',
        fontSize: 20,
        marginVertical: 20


    },
    register: {
        fontFamily: 'PixeloidsSanBold',
        fontSize: 20,
        marginVertical: 5
    }
})



