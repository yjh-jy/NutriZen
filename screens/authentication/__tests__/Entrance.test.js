import renderer from 'react-test-renderer';
import Entrance from '../Entrance';

describe('<Entrance />', () => {

    test('renders correctly', ()=>{
        const tree = renderer
        .create(<Entrance/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
    });

})