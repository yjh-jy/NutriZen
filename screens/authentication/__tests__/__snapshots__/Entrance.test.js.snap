// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Entrance /> renders correctly 1`] = `null`;
